File: student-information-system-backend/src/main/java/com/trkgrn/studentinformationsystem/api/model/entity/File.java
package com.trkgrn.studentinformationsystem.api.model.entity;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;

@Embeddable
@AllArgsConstructor
@NoArgsConstructor
@Data
public class File {

    @Column(name = "file_name")
    private String fileName;

    @Column(name = "file_extension")
    private String fileExtension;

    @Column(name = "content_type")
    private String contentType;

    @Column(name = "file_size")
    private Long fileSize;

    @Column(name = "cloud_path")
    private String cloudPath;

    @Column(name = "download_url")
    private String downloadUrl;

}

File: student-information-system-frontend/src/app/models/entity/File.ts
export class File {
  fileName?: string;
  fileExtension?: string;
  contentType?: string;
  fileSize?: number;
  cloudPath?: string;
  downloadUrl?: string;

  constructor(fileName: string, fileExtension: string, contentType: string, fileSize: number, cloudPath: string, downloadUrl: string) {
    this.fileName = fileName;
    this.fileExtension = fileExtension;
    this.contentType = contentType;
    this.fileSize = fileSize;
    this.cloudPath = cloudPath;
    this.downloadUrl = downloadUrl;
  }
}
